auth0/auth0-authorization-extension
# Remove hapi-swagger, use static json

## ✏️ Changes
  
- `hapi-swagger` was causing an issue with conflicting versions of `joi`, which it requires as a subdependency vs the version of `joi` used by the main project.
- this PR removes `hapi-swagger` as a dependency, instead storing it's output as a static json file in the repo. This is deemed to be suitable as there is no intention to modify the API of the extension in future, but it should be noted that if we ever did, we'd need to manually update that JSON file in order to update the API explorer.
  
## 📷 Screenshots

- Tested in `prod-us-4` using `joi@17.12.2` 
![image](https://github.com/user-attachments/assets/31bfdd56-edef-423e-963d-2bf0e507eaf8)

- Test in `prod-jp-1` using `joi@17.12.2` 
![image](https://github.com/user-attachments/assets/e1bb98ae-4257-42ae-ac4c-4c6a533ca8d9)


  
## 🔗 References
  
- Jira: https://auth0team.atlassian.net/browse/IDS-5232
  
## 🎯 Testing
  
- Follow the instructions [documented elsewhere](https://oktawiki.atlassian.net/wiki/spaces/Bacca/pages/3103917188/Authorization+Extension+Upgrade+Deployment+Plan) to build the extensions locally with the dev branch of auth0-exentsions-cli
- upload to `/extensions/deploy` using the extensions-deployment tool
- install the authz-extension in a prod tenant using the chrome overrides hack for extensions.json and extensions-gallery
- open the extension and use chrome overrides to override the html page to replace the path of the assets with `/extensions/develop`
- navigate to the API explorer and check if it works or errors
   
✅ This change has been tested in a Webtask
 
🚫 This change has unit test coverage
  
🚫 This change has integration test coverage
  
🚫 This change has been tested for performance
  
## 🚀 Deployment
  
✅ This can be deployed any time
  
## 🎡 Rollout
  
In order to verify that the deployment was successful we will deploy the extensions in `/extensions/develop` and test in a prod tenant
  
## 🔥 Rollback
  
We will rollback if we encounter any issues
  
### 📄 Procedure
  
- if the extension update is released, we'll follow the [recovery plan](https://oktawiki.atlassian.net/wiki/spaces/Bacca/pages/3115255000/Authorization+Extension+Update+Disaster+Recovery+Plan)
